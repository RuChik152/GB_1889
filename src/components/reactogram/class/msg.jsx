import React, { Component } from 'react';

export class Msg extends Component {
  render() {
    return (
      <div className="msg__item">
        <span className="item__author">--{this.props.item.author}--</span>
        <p className="item__text">--{this.props.item.msg}--</p>
        <p className="item__time">--{this.props.item.time}--</p>
      </div>
    );
  }
}
