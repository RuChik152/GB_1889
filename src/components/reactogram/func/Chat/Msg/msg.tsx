import React, { FC } from 'react';
import style from './msg.module.scss';

interface MsgProps {
  key: string;
  author: string;
  msg: string;
  time: string;
}

export const Msg: FC<MsgProps> = ({ author, msg, time }) => {
  return (
    <div className={style.msg__item}>
      <span className={style.item__author} data-testid="test-author">
        {author}
      </span>
      <p className={style.item__text} data-testid="test-text">
        {msg}
      </p>
      <p className={style.item__time} data-testid="test-time">
        Время: {time}
      </p>
    </div>
  );
};
