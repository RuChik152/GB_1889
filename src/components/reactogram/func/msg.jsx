import React from "react";

export const Msg = (props) => {

        return  <div className="msg__item">
                    <span className="item__author">--{ props.item.author }--</span>
                    <p className="item__text">--{ props.item.msg }--</p>
                    <p className="item__time">--{ props.item.time }--</p>
                </div>

}