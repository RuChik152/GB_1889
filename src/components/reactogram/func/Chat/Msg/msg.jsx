import React from 'react';
import './msg.scss'

export const Msg = (props) => {
  return (
    <div className="msg__item">
      <span className="item__author" data-testid="test-author">
        {props.item.author}
      </span>
      <p className="item__text" data-testid="test-text">
        {props.item.msg}
      </p>
      <p className="item__time" data-testid="test-time">
        Время: {props.item.time}
      </p>
    </div>
  );
};
