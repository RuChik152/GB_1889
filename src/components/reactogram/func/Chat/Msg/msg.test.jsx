import React from 'react';
import { render, screen } from '@testing-library/react';
import { Msg } from './msg';
import '@testing-library/jest-dom';

const data = {
  author: 'admin',
  msg: 'Тестовое сообщение!',
  time: '10:10:10',
};

describe('Msg', () => {
  it('render component', () => {
    render(<Msg item={data} />);
  });

  it('text on elements', () => {
    render(<Msg item={data} />);
    expect(screen.getByTestId('test-author').innerHTML).toBe('admin');
    expect(screen.getByTestId('test-text').innerHTML).toBe(
      'Тестовое сообщение!'
    );
    expect(screen.getByTestId('test-time').innerHTML).toBe('Время: 10:10:10');
  });

  it('Snapshot test', () => {
    const { asFragment } = render(<Msg item={data} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
