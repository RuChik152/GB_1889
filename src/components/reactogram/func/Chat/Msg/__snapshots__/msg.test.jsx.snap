// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Msg Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="msg__item"
  >
    <span
      class="item__author"
      data-testid="test-author"
    >
      admin
    </span>
    <p
      class="item__text"
      data-testid="test-text"
    >
      Тестовое сообщение!
    </p>
    <p
      class="item__time"
      data-testid="test-time"
    >
      Время: 10:10:10
    </p>
  </div>
</DocumentFragment>
`;
